cvereporter/nist_enhance.py

```diff
@@ -0,0 +1,74 @@
+from cyclonedx.model.impact_analysis import ImpactAnalysisAffectedStatus
+from cyclonedx.model.vulnerability import Vulnerability, VulnerabilitySource,VulnerabilityScoreSource, VulnerabilityRating, VulnerabilitySeverity, BomTarget, BomTargetVersionRange
+import requests
+import json
+
+def fetch_nist(url: str, id: str) -> dict:
+    data = None
+    nist_resp = requests.get(url)
+    if nist_resp.status_code != 200:
+        print("error fetching "+id)
+    else:
+        data = nist_resp.json()
+        with open("data/nist_"+id+".json", "w") as dest:
+            json.dump({"url":url, "data": data}, dest, indent=True)
+    return data
+def extract_relevant_parts(nist_resp: dict) -> dict:
+    # todo: this can use a unit test at some point
+    resp_dict = {}
+    ratings = []
+    cve = nist_resp["vulnerabilities"][0]["cve"]
+    #todo: do we have more than 1 cve in a resp?
+    description = ""
+    for desc in cve["descriptions"]:
+        if(desc["lang"] == "en"):
+            description = desc["value"]
+
+    for metrics in cve["metrics"]["cvssMetricV31"]:
+        #todo: do we need recommendations from NIST as well?
+        relevant = {}
+        relevant["source"] = metrics["source"]
+        relevant["score"] = metrics["cvssData"]["baseScore"]
+        relevant["severity"] = metrics["cvssData"]["baseSeverity"]
+        relevant["method"] = "CVSSv3" #is this always true?
+        relevant["vector"] = metrics["cvssData"]["vectorString"]
+        ratings.append(relevant)
+    resp_dict["ratings"] =  ratings
+    resp_dict["description"] = description
+    resp_dict["versions"] = extract_versions(cve["configurations"])
+    return resp_dict
+def extract_versions(cve_configs):
```

Comment:
@tellison i think this is dubious, but i don't have a better way of finding an affects version. I'm open to suggestions here (i've basically manually parsed out the oracle jdk version, minus update, but we can special case that, and i'm assuming it's 1:1 with open jdk). There's code to extract it from openjvg, but they publish it at the top of the webpage, and the webpage can contain multiple cves, so i'm not sure that's the best place to get information